tutorials-v5/quantum-circuits/v5_paper-qip2.md

```diff
@@ -0,0 +1,367 @@
+
+# exp(-i XX t)
+trotter_simulation_noisy.add_gate("H", targets=[2])
+trotter_simulation_noisy.add_gate("RZX", targets=[2, 3], arg_value=sqrt_term)
+trotter_simulation_noisy.add_gate("H", targets=[2])
+
+# exp(-i YY t)
+trotter_simulation_noisy.add_gate("RZ", 3, arg_value=np.pi / 2)
+trotter_simulation_noisy.add_gate("RX", 2, arg_value=-np.pi / 2)
+trotter_simulation_noisy.add_gate("RZX", [2, 3], arg_value=sqrt_term)
+trotter_simulation_noisy.add_gate("RZ", 3, arg_value=-np.pi / 2)
+trotter_simulation_noisy.add_gate("RX", 2, arg_value=np.pi / 2)
+
+trotter_simulation_noisy.draw("matplotlib")
+```
+
+```python
+state_system = ket2dm(init_state)
+state_trotter_circ = [init_state]
+ancilla = basis(2, 1) * basis(2, 1).dag()
+for dd in range(num_steps):
+    state_full = tensor(ancilla, state_system, ancilla)
+    state_full = trotter_simulation_noisy.run(state=state_full)
+    state_system = state_full.ptrace([1, 2])
+    state_trotter_circ.append(state_system)
+```
+
+Again, we want to run this trotterized evolution on the suuperconducting hardware backend. Be aware that, due of the increased complexity, this computation can take several minutes depending on your hardware.
+
+```python
+processor = SCQubits(num_qubits=4, t1=3.0e4, t2=3.0e4)
+processor.load_circuit(trotter_simulation_noisy)
+```
+
+```python
+state_system = ket2dm(init_state_trit)
+state_list_proc = [state_system]
+for dd in range(num_steps):
+    state_full = tensor(
+        basis(3, 1) * basis(3, 1).dag(),
+        state_system,
+        basis(3, 1) * basis(3, 1).dag(),
+    )
+    result_noisey = processor.run_state(
+        state_full,
+        solver="mesolve",
+        options={
+            "store_states": False,
+            "store_final_state": True,
+        },
+    )
+    state_full = result_noisey.final_state
+    state_system = state_full.ptrace([1, 2])
+    state_list_proc.append(state_system)
+    print(f"Step {dd+1}/{num_steps} finished.")
+```
+
+```python
+processor.plot_pulses()
+```
+
+### Comparison of Results
+
+```python
+# Standard mesolve solution
+sm1 = tensor(destroy(2).dag(), qeye(2))
+sm2 = tensor(qeye(2), destroy(2).dag())
+c_ops = [np.sqrt(gam) * sm1, np.sqrt(gam) * sm2]
+
+result_me = mesolve(H, init_state, tlist, c_ops, e_ops=[sz1, sz2])
+```
+
+```python
+plt.plot(tlist, result_me.expect[0], "-", label=r"Ideal")
+plt.plot(times_circ, expect(sz1, state_trotter_circ), "--d", label="trotter")
+plt.plot(
+    times_circ,
+    expect(sz_qutrit & qeye(3), state_list_proc),
+    "-.o",
+    label=r"noisy hardware",
+)
+plt.xlabel("Time")
+plt.ylabel("Expectation values")
+plt.legend()
+plt.show()
+```
+
+## References
+
+\[1\] [Li, et. al, Quantum (2022)](http://dx.doi.org/10.22331/q-2022-01-24-630)
+
+\[2\] [Ding, et. al, PRX Quantum (2024)](https://doi.org/10.1103/PRXQuantum.5.020332)
+
+\[3\] [Cleve and Lang, ICALP (2017)](https://doi.org/10.48550/arXiv.1612.09512)
+
+
+## About
+
+```python
+about()
+```
+
+## Testing
+
+```python
+# TODO
```

Comment:
testing, hmm, am not super sure, since we design these circuits to have finite error, so none of the ''results'' will match perfectly.